### PR TITLE
code and results of TDNN-PGRU

### PR DESCRIPTION
**Pull request by GaofengCheng**
This PR is not for merging at this stage,
just showing how I implement projected-GRU (TDNN-PGRU) and the results of it for @danpovey

### CODE REVIEW

**Comment by GaofengCheng** [comment]
@danpovey This PR is related the model we discussed in email, I have not had time to make a clean PR, you can focus on lstm.py and the modifications of shell scripts

File: egs/ami/s5b/local/chain/tuning/run_tdnn_lstm_1j.sh [hunk 1]
```diff
@@ -20,6 +20,14 @@
 # exp/sdm1/chain_cleaned/tdnn_lstm1i_sp_bi_ihmali_ld5/: num-iters=87 nj=2..12 num-params=43.4M dim=40+100->3770 combine=-0.142->-0.131 xent:train/valid[57,86,final]=(-1.78,-1.48,-1.48/-2.22,-2.17,-2.16) logprob:train/valid[57,86,final]=(-0.157,-0.117,-0.114/-0.243,-0.249,-0.245)
 # exp/sdm1/chain_cleaned/tdnn_lstm1j_sp_bi_ihmali_ld5/: num-iters=87 nj=2..12 num-params=43.4M dim=40+100->3770 combine=-0.139->-0.130 xent:train/valid[57,86,final]=(-1.82,-1.50,-1.48/-2.18,-2.12,-2.11) logprob:train/valid[57,86,final]=(-0.165,-0.121,-0.119/-0.240,-0.247,-0.246)
 
+# System               tdnn_gru1a_sp_bi_ihmali_ld5tdnn_lstm1j_sp_bi_ihmali_ld5
```
Comment (jtrmal) [inline egs/ami/s5b/local/chain/tuning/run_tdnn_lstm_1j.sh]: is there a space missing between the names of the experiments or is it just github rendering issue?

File: egs/wsj/s5/steps/libs/nnet3/xconfig/parser.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
         'fixed-affine-layer' : xlayers.XconfigFixedAffineLayer,
         'idct-layer' : xlayers.XconfigIdctLayer,
         'affine-layer' : xlayers.XconfigAffineLayer,
+        'gru-layer' : xlayers.XconfigGRULayer,
```
Comment (jtrmal) [inline egs/wsj/s5/steps/libs/nnet3/xconfig/parser.py]: Should it be GRULayer or GruLayer?

File: egs/ami/s5b/local/chain/tuning/run_tdnn_lstm_1j.sh (hunk 1 above)
Comment (GaofengCheng) [inline egs/ami/s5b/local/chain/tuning/run_tdnn_lstm_1j.sh]: @jtrmal oops it's the ./local/chain/compare_wer_general.sh 's problem, I will fix and make a PR. Thx !

File: egs/wsj/s5/steps/libs/nnet3/xconfig/parser.py (hunk 2 above)
Comment (GaofengCheng) [inline egs/wsj/s5/steps/libs/nnet3/xconfig/parser.py]: GruLayer should be better

**Pull request closed without merging**